### DISCUSSION TITLE
Running python on Xcode 12/13?

### DISCUSSION THREAD

**Discussion by stardriftfx** [UWAppDev/community#10] [Q&A] [answered]
Is there a way to run python on Xcode? Python 3.x preferred

**Comment by personalizedrefrigerator** [comment]
Do you want to use Xcode as an editor for python? If so, I'm not sure...

If you want to use Python in an iOS app, you may want to look at the [`ios_system`](https://github.com/holzschu/ios_system.git) library.

Unfortunately, it currently only documents how to use `python2`. Apps using the library seem to have access to `python3`, though. See the [build instructions for a project that uses `ios_system`](https://github.com/holzschu/a-shell#how-to-compile-it) for more information. Note that the build instructions for the linked project (a-Shell) don't work for me, but I don't have much experience with building projects on a mac.

**Reply by ApolloZhu** [reply]
Since we are on the topics of Python on iOS, here's a Python3 IDE for iOS: http://omz-software.com/pythonista/

**Comment by Lucas-ZX-W** [comment, marked as answer]
I've used Xcode for python before, but I used it in combination with the command line to activate the interpreter so really I was just using Xcode as a fancy TextEdit. There is a limited amount of syntax highlighting and auto-complete (compared to something like PyCharm, for example) but it's good enough (I think) for certain programs (e.g Raspberry Pi). BUT, there is a way to set up Xcode with a scheme that runs the python program inside the built-in console with the run button/cmd + R, I found [this blog post](https://ericasadun.com/2016/12/04/running-python-in-xcode-step-by-step/) that @ApolloZhu dug up to be fairly constructive.
